api/api_request_logger.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	log "github.com/inconshreveable/log15"
+)
+
+// RequestLogger logs requests to an output
+type RequestLogger struct {
+	enabled      bool
+	writerChan   chan entry
+	stopChan     chan bool
+	outputWriter func(msg string, ctx ...interface{})
+}
+
+// NewThorRequestLogger applies thor business logic
+// when returning an instance of *RequestLogger
+func NewThorRequestLogger(enabledFlag bool, enableLog bool) *RequestLogger {
```

Comment:
This looks wired to me

Reply:
Perhaps this might look weird because this type of business logic used to be existing in files like the `utils.go`. 

Arguably, files like `utils.go` should not hold the thor business logic on how to instantiate a package.
It make it hard to test the logic and the most important instantiation logic is pushed into a gather-all kind of file.
ex. The [flags around the p2p](https://github.com/vechain/thor/blob/master/cmd/thor/utils.go#L446-L496)

Typically, the way that I've seen this done is by having the package implement a `NewThorBusinessLogicX` that holds the business logic. That way it's possible to create unit tests for the package AND for the business logic.